Detect the FAILED state of a display at finish time, not at unmanage 
time. Otherwise it would be set as FINISHED first and treated like it 
ended properly.

Also bring the delay down to 2 seconds.

Index: gdm3-3.4.1/daemon/gdm-display.c
===================================================================
--- gdm3-3.4.1.orig/daemon/gdm-display.c	2013-02-10 17:28:37.003755377 +0100
+++ gdm3-3.4.1/daemon/gdm-display.c	2013-02-10 17:29:02.819881581 +0100
@@ -665,11 +665,21 @@ gdm_display_manage (GdmDisplay *display)
 static gboolean
 gdm_display_real_finish (GdmDisplay *display)
 {
+        gdouble elapsed;
+
         g_return_val_if_fail (GDM_IS_DISPLAY (display), FALSE);
 
-        _gdm_display_set_status (display, GDM_DISPLAY_FINISHED);
+        g_timer_stop (display->priv->slave_timer);
+        elapsed = g_timer_elapsed (display->priv->slave_timer, NULL);
+
+        if (elapsed < 2) {
+                g_warning ("GdmDisplay: display lasted %lf seconds", elapsed);
+                _gdm_display_set_status (display, GDM_DISPLAY_FAILED);
+        } else {
+                _gdm_display_set_status (display, GDM_DISPLAY_FINISHED);
+        }
 
-        g_debug ("GdmDisplay: finish display");
+        g_debug ("GdmDisplay: finished display after %lf seconds", elapsed);
 
         return TRUE;
 }
@@ -693,14 +703,10 @@ gdm_display_finish (GdmDisplay *display)
 static gboolean
 gdm_display_real_unmanage (GdmDisplay *display)
 {
-        gdouble elapsed;
-
         g_return_val_if_fail (GDM_IS_DISPLAY (display), FALSE);
 
         g_debug ("GdmDisplay: unmanage display");
 
-        g_timer_stop (display->priv->slave_timer);
-
         if (display->priv->slave_proxy != NULL) {
                 gdm_slave_proxy_stop (display->priv->slave_proxy);
 
@@ -720,13 +726,7 @@ gdm_display_real_unmanage (GdmDisplay *d
                 display->priv->access_file = NULL;
         }
 
-        elapsed = g_timer_elapsed (display->priv->slave_timer, NULL);
-        if (elapsed < 3) {
-                g_warning ("GdmDisplay: display lasted %lf seconds", elapsed);
-                _gdm_display_set_status (display, GDM_DISPLAY_FAILED);
-        } else {
-                _gdm_display_set_status (display, GDM_DISPLAY_UNMANAGED);
-        }
+        _gdm_display_set_status (display, GDM_DISPLAY_UNMANAGED);
 
         return TRUE;
 }
